src/databricks/labs/lsql/dashboards.py

```diff
@@ -199,40 +281,29 @@ def _parse_dashboard_metadata(dashboard_folder: Path) -> DashboardMetadata:
             logger.warning(f"Parsing {dashboard_metadata_path}: {e}")
             return fallback_metadata
 
-    def _parse_widget_metadata(self, path: Path, widget: Widget, order: int) -> WidgetMetadata:
-        width, height = self._get_width_and_height(widget)
-        fallback_metadata = WidgetMetadata(
-            order=order,
-            width=width,
-            height=height,
-        )
-
-        try:
-            parsed_query = sqlglot.parse_one(path.read_text(), dialect=sqlglot.dialects.Databricks)
-        except sqlglot.ParseError as e:
-            logger.warning(f"Parsing {path}: {e}")
-            return fallback_metadata
-
-        if parsed_query.comments is None or len(parsed_query.comments) == 0:
-            return fallback_metadata
-
-        first_comment = parsed_query.comments[0]
-        return fallback_metadata.replace_from_arguments(shlex.split(first_comment))
+    def _get_widget(self, widget_metadata: WidgetMetadata) -> Widget:
+        if widget_metadata.spec_type == MarkdownSpec:
+            return self._get_text_widget(widget_metadata)
+        return self._get_counter_widget(widget_metadata)
 
     @staticmethod
-    def _get_text_widget(path: Path) -> Widget:
-        widget = Widget(name=path.stem, textbox_spec=path.read_text())
+    def _get_text_widget(widget_metadata: WidgetMetadata) -> Widget:
+        widget = Widget(
```

Comment:
Don't you want to create Widget instance from a method of Tile instance?.. eg turning "abstract tiles" into "concrete widgets"

Reply:
yes, I like that idea. I am planning to do that in two PRs, this and #150. The tile's make more sense when introducing more specs. I'll rework the other PR on top of this one, so that we can discuss EOD